docs/specs/idp-account-linking.md

```diff
@@ -12,47 +12,60 @@ Here is an example of the configuration:
 
 ```yaml
 identity:
-  providers:
-    - alias: google
-      client_id: exampleclientid
-      type: google
-      account_linking:
-        enabled: true
-        idp_claim_key: "email"
-        match_against_claim_key: "email"
+  on_conflict:
+    account_linking:
+      oauth: "merge"
```

Comment:
I guess the key `oauth` means From OAuth. That is, this config is about the scenario that the triggering identity is an OAuth Identity.

This config, I believe, can handle the following cases

1. Triggering=OAuth, existing=OAuth
2. Triggering=OAuth, existing=Login ID

Then, I wonder why do we need to separate the config of From OAuth and From Login ID, because if we did not make the distinction, the cases would be

1. Triggering=OAuth, existing=OAuth
2. Triggering=OAuth, existing=Login ID
3. Triggering=Login ID, existing=OAuth (The handling of this case should be the same as case 1)

Reply:
> Triggering=Login ID, existing=OAuth (The handling of this case should be the same as case 1)

I doubt if it is the case. We probably want to block and ask the user to login with oauth if this is this case, while we want to automatically merge accounts in case 1 and 2.

Reply:
Proposed change:

```
identity:
  on_conflict:
    signup:
      oauth: "merge" # "error" | "merge"
      login_id: "error" # "error" is the only supported value at the moment.
```